### DISCUSSION TITLE
Hover doesn't apply styles in Firefox if d3.select.raise is used.

### DISCUSSION THREAD

**Discussion by srikant-ch5** [d3/d3#3910] [Q&A] [answered]
Issue Summary: I'm encountering a problem where hover styles are applied differently across browsers. Specifically, when using d3.select.raise() in conjunction with a hover event, the styles are correctly applied in Chrome but fail to apply in Firefox.

Issue Details:

Objective: Apply specific styles to an element when it is hovered over.
Problem: The hover styles work as expected in Chrome but do not get applied in Firefox when using d3.select.raise().
Demo: Below is a demonstration of the issue, highlighting the inconsistent behavior across browsers.
[demo](https://jsfiddle.net/zt54mw8c/)

**Comment by Fil** [comment]
d3 v4 is an ancient version. Also if you have further questions please open a [discussion](https://github.com/d3/d3/discussions/categories/q-a) instead of an issue.

**Reply by srikant-ch5** [reply]
Please refer below demo I have updated it with v7.
https://jsfiddle.net/2t9mzkaj/

**Reply by Fil** [reply, marked as answer]
Thanks. This looks like a bug with Firefox, as you can see if you inspect the DOM while you mouse over the paths, D3 is doing the right thing.

**Reply by Fil** [reply]
To make it work in Firefox you could add a `hover` class:

```
    svg.selectAll("g").on("mouseover", function(){
      d3.select(this).raise().classed("hover", true)

    });
    svg.selectAll("g").on("mouseout", function(){
        d3.select(this).lower().classed("hover", false)
    });
```

then change `g:hover` to `g.hover` in your css.
